RATapi/utils/plotting.py

```diff
@@ -224,3 +303,526 @@ def __exit__(self, _exc_type, _exc_val, _traceback):
+        axs[i].set_visible(False)
+
+    fig.tight_layout()
+    return fig
+
+
+@assert_bayesian("Histogram")
+def plot_hists(
+    results: RATapi.outputs.BayesResults,
+    params: Union[list[Union[int, str]], None] = None,
+    smooth: bool = True,
+    sigma: Union[float, None] = None,
+    estimated_density: Union[
+        dict[Literal["normal", "lognor", "kernel", None]], Literal["normal", "lognor", "kernel", None]
+    ] = None,
+    block: bool = False,
+    return_fig: bool = False,
+    **hist_settings,
+):
+    """Plot marginalised posteriors for several parameters from a Bayesian analysis.
+
+    Parameters
+    ----------
+    results : BayesResults
+        The results from a Bayesian calculation.
+    params : list[int], default None
+        The indices or names of a subset of parameters if required.
+        If None, uses all indices.
+    smooth : bool, default True
+        Whether to apply a Gaussian smoothing to the histogram.
+        Defaults to True.
+    sigma: float or None, default None
+        If given, is used as the sigma-parameter for the Gaussian smoothing.
+        If None, the default (1/3rd of parameter chain standard deviation) is used.
+    estimated_density : dict, default None
+        If None (default), ignore.
+        Can also be a string 'normal', 'lognor' or 'kernel' to apply the same estimated density to all parameters.
+        Else, a dictionary where the keys are
+        indices or names of parameters, and values denote an estimated density
+        of the given form on top of the histogram:
+        None : do not plot estimated density for this parameter.
+        'normal': normal Gaussian.
+        'lognor': Log-normal probability density.
+        'kernel': kernel density estimation.
+        To provide a default estimated density function to all parameters that haven't been specifically set,
+        pass the 'default' key,
+        e.g. to apply 'normal' to all unset parameters, set `estimated_density = {'default': 'normal'}`.
+    block : bool, default False
+        Whether Python should block until the plot is closed.
+    return_fig: bool, default False
+        If True, return the figure as an object instead of showing it.
+    hist_settings :
+        Settings passed to `np.histogram`. By default, the settings
+        passed are `bins = 25` and `density = True`.
+
+    Returns
+    -------
+    Figure or None
+        If `return_fig` is True, return the figure - otherwise, return nothing.
+
+    """
+
+    # first convert names to indices if given
+    fitname_to_ind = partial(name_to_index, names=results.fitNames)
+
+    if params is None:
+        params = range(0, len(results.fitNames))
+    else:
+        params = list(map(fitname_to_ind, params))
+
+    if estimated_density is not None:
+        if isinstance(estimated_density, str):
+            estimated_density = {"default": estimated_density}
+        default = estimated_density.get("default")
```

Comment:
can't remove lines 716-717 because `pop` throws a `KeyError` if there is no key `default`

Reply:
changed `get` to `pop` but you can't remove `estimated_density = {"default": estimated_density}` as if `estimated_density` is given as a string, you can't iterate over its items and the iterator breaks

Reply:
can't do this because you iterate over `estimated_density.items()` either. although looking at it i'm becoming dissatisfied with how the loop looks generally. will try to come up with a neater way